### DISCUSSION TITLE
AirCaption: Subtitle video on Windows & Mac with Whisper.cpp

### DISCUSSION THREAD

**Discussion by kevin2379** [ggml-org/whisper.cpp#1065] [Show and tell]
[AirCaption](https://www.aircaption.com/?utm_source=github&utm_medium=discussion&utm_campaign=whispercpp-show-tell) allows users to run whisper.cpp locally on Windows & Mac to generate and edit captions for video. 

[https://github.com/ggerganov/whisper.cpp/assets/64876105/8f8d4d9a-16dc-426e-913f-c8ce60fb9127](https://github.com/ggerganov/whisper.cpp/assets/64876105/8f8d4d9a-16dc-426e-913f-c8ce60fb9127)

### Features

- Subtitle video or audio in multiple languages
- Multiple whisper.cpp models (tiny to large)
- Easily edit text and timing of captions
- Merge and split captions
- Import and edit existing caption files
- Export SRT, VTT or TXT files

### Technologies
- Whisper.cpp
- Electron
- React
- FFMPEG
- Video.js
- Wavesurfer.js

The software was originally intended for video content producers but seems to be more popular with content consumers who want to caption videos themselves before watching (when subtitles don’t already exist). 

Download for free at [https://www.aircaption.com/download](https://www.aircaption.com/download?utm_source=github&utm_medium=discussion&utm_campaign=whispercpp-show-tell)

Thank you [@ggerganov](https://github.com/ggerganov) for all the work you’ve put into whisper.cpp. This project would not have been possible without you!

**Comment by albertputinjobs** [comment]
Can u make a chrome extension which can display live subtitles on top of video ? that would be great!

**Reply by lightninglu10** [reply]
@albertputinjobs I'm building that right now! Might be done in about 2 weeks or so: 


https://github.com/ggerganov/whisper.cpp/assets/2453737/c718dfc1-bd63-4388-adb3-231eee325031

**Reply by kevin2379** [reply]
@albertputinjobs I think that’s a great idea! Glad to hear @lightninglu10 has already started on that. To run in browser, this would likely need to use the tiny.en or possibly base.en models, and those should work fine for this purpose. 

AirCaption is intended more for creating very accurate SRT and VTT files that can be distributed along with videos online. For example, a user could use the medium model which would take longer but might be around 90% accurate, then they can edit to get to 99% accuracy, which is what would typically be required for this purpose. 

The next feature on the AirCaption roadmap is support for multiple video projects and a transcribe Queue, so that users can set many videos to transcribe and they will all get done one after another.
